internal/request/graphql/schema/collection.go

```diff
@@ -345,6 +369,35 @@ func indexFromAST(directive *ast.Directive) (client.IndexDescription, error) {
 	return desc, nil
 }
 
+func defaultFromAST(
+	field *ast.FieldDefinition,
+	directive *ast.Directive,
+) (any, error) {
+	astNamed, ok := field.Type.(*ast.Named)
+	if !ok {
+		return nil, NewErrDefaultValueNotAllowed(field.Name.Value, field.Type.String())
+	}
+	var value any
+	for _, arg := range directive.Arguments {
+		if defaultPropNameToType[arg.Name.Value] != astNamed.Name.Value {
+			return nil, NewErrDefaultValueInvalid(astNamed.Name.Value, arg.Name.Value)
```

Comment:
todo: Please add tests for this error condition

Reply:
done